modules/serverless-kafka-broker-tls-custom-config.adoc

```diff
@@ -0,0 +1,75 @@
+// Module is included in the following assemblies:
+//
+// * serverless/knative_eventing/serverless-kafka.adoc
+
+[id="serverless-kafka-broker-tls-custom-config_{context}"]
+= Configuring TLS authentication for the Kafka Broker
+
+.Prerequisites
+
+* A Kafka cluster CA certificate as a `.pem` file.
+* A Kafka cluster client certificate and key as `.pem` files.
+
+.Procedure
+
+. Create the certificate files as secrets in your chosen namespace:
++
+[source,terminal]
+----
+$ oc create secret --namespace <namespace> generic <kafka_auth_secret> \
+  --from-literal=protocol=SSL \
+  --from-file=ca.crt=caroot.pem \
+  --from-file=user.crt=certificate.pem \
+  --from-file=user.key=key.pem
+----
++
+[IMPORTANT]
+====
+Use the key names `ca.crt`, `user.crt`, and `user.key`. Do not change them.
+====
+
+. Create a `ConfigMap` for configuration as a YAML file:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-kafka-broker-config
+data:
+  default.topic.partitions: <partitions>
+  default.topic.replication.factor: <replication_factor>
+  bootstrap.servers: <bootstrap_servers>
+  auth.secret.ref.name: <kafka_auth_secret>
+----
+
+. Apply the `ConfigMap` YAML file:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. R
+
+. Reference your the `ConfigMap` on your `Kafka` broker:
++
+[source,yaml]
+----
+apiVersion: eventing.knative.dev/v1
+kind: Broker
+metadata:
+  annotations:
+    eventing.knative.dev/broker.class: Kafka
+  name: my-demo-kafka-broker
+spec:
+  config:
+    apiVersion: v1
+    kind: ConfigMap
+    name: my-kafka-broker-config
+    namespace: <namespace>
+----
+
+.Additional resources
+
+* link:https://access.redhat.com/documentation/en-us/red_hat_amq/7.5/html-single/using_amq_streams_on_rhel/index#assembly-kafka-encryption-and-authentication-str[TLS and SASL on Kafka]
```

Comment:
Rather than duplicate this for different modules, maybe just add an additional resources section to the assembly and add this there?

Reply:
It's basically the same as in the `modules/serverless-kafka-tls.adoc` file, which I used as inspiration.